Author: Javi Merino <vicho@debian.org>
Description: Set the group of the test directory to one of the user's group
  Some $TMPDIRs create folders owned by a group which the user doesn't
  belong to.  test-inherit-mode then fails when chmod tries to set the
  setgid bit because the user doesn't have permission to do so.  This
  patch sets the group of the parent directory to one of the users'
  groups, so that the test can modify the setgid group of any of the
  directories created inside it afterwards.
Forwarded: http://selenic.com/pipermail/mercurial-devel/2012-March/038419.html
--- a/tests/test-inherit-mode.t
+++ b/tests/test-inherit-mode.t
@@ -8,6 +8,7 @@
 just in case somebody has a strange $TMPDIR
 
   $ chmod g-s dir
+  $ chgrp $(groups | awk '{print $1}') dir
   $ cd dir
 
   $ cat >printmodes.py <<EOF
